ibis/expr/operations.py

```diff
@@ -1515,11 +1529,25 @@ def _get_schema(self):
         sright = right.schema()
 
         overlap = set(sleft.names) & set(sright.names)
+
+        deletes = []
+        for predicate in self.predicates:
+            op = predicate.op()
+            if (
+                isinstance(op, Equals) and
+                op.left.get_name() == op.right.get_name() and
+                op.left.get_name() in overlap
+            ):
+                overlap.remove(op.left.get_name())
+                deletes.append(op.left.get_name())
```

Comment:
This can only be done for `INNER JOIN`. When you use a `LEFT JOIN` (for example), the choice of key matters if it overlaps.

```sql
postgres=# create temp table t as select generate_series(1, 10) as t;
postgres=# create temp table s as select generate_series(1, 5) as t;
postgres=# select t.t as t_t, s.t as s_t from t left join s using (t);
 t_t | s_t 
-----+-----
   1 |   1
   2 |   2
   3 |   3
   4 |   4
   5 |   5
   6 |    
   7 |    
   8 |    
   9 |    
  10 |    
(10 rows)
```

Reply:
Indeed, it needs more tests too.